subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/support/decorators/DependencyHandlerDecorator.kt

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.kotlin.dsl.support.decorators
+
+import groovy.lang.Closure
+
+import org.gradle.api.Action
+import org.gradle.api.artifacts.Dependency
+import org.gradle.api.artifacts.dsl.ComponentMetadataHandler
+import org.gradle.api.artifacts.dsl.ComponentModuleMetadataHandler
+import org.gradle.api.artifacts.dsl.DependencyConstraintHandler
+import org.gradle.api.artifacts.dsl.DependencyHandler
+import org.gradle.api.artifacts.query.ArtifactResolutionQuery
+import org.gradle.api.artifacts.transform.TransformAction
+import org.gradle.api.artifacts.transform.TransformParameters
+import org.gradle.api.artifacts.transform.TransformSpec
+import org.gradle.api.artifacts.transform.VariantTransform
+import org.gradle.api.artifacts.type.ArtifactTypeContainer
+import org.gradle.api.attributes.AttributesSchema
+
+
+abstract class DependencyHandlerDecorator : DependencyHandler {
```

Comment:
It would be great to add a brief note and maybe a link to some more detailed explanation to these classes as to why we need to have them.

Reply:
The `package-info.java` could be a good place to document how this works.

Reply:
`package-info.java` has no equivalent in Kotlin currently, I'll use `GradleDecorator` as the entry point and link from the other decorators.